x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypes.java

```diff
@@ -67,6 +69,9 @@ public static DataType fromJava(Object value) {
         if (value instanceof Short) {
             return SHORT;
         }
+        if (value instanceof OffsetTime) {
```

Comment:
`TIME` should be after not before `ZonedDateTime` since it's a rare type (see `Interval` which is at the bottom).

Reply:
Same here